ts/features/bonus/siciliaVola/components/SvVoucherListFilters.tsx

```diff
@@ -0,0 +1,241 @@
+  >
+    <H4 weight={checked ? "SemiBold" : "Regular"} color={"bluegreyDark"}>
+      {text}
+    </H4>
+    <IconFont
+      name={checked ? "io-radio-on" : "io-radio-off"}
+      size={22}
+      color={checked ? IOColors.blue : IOColors.bluegrey}
+    />
+  </ListItem>
+);
+
+const SvVoucherListFilters: React.FunctionComponent<Props> = (props: Props) => {
+  const [searchValue, setSearchValue] = useState<string | undefined>(undefined);
+  const [departureDate, setDepartureDate] = useState<Date | undefined>(
+    undefined
+  );
+  const [returnDate, setReturnDate] = useState<Date | undefined>(undefined);
+
+  const [voucherStatus, setVoucherStatus] = useState<number | undefined>(
+    undefined
+  );
+
+  const onVoucherStatusSelect = (value: number) => {
+    setVoucherStatus(value);
+    Keyboard.dismiss();
+  };
+
+  const renderVoucherStateItem = (
+    listItem: ListRenderItemInfo<StatoVoucherBean>
+  ) =>
+    listItem.item.idStato && listItem.item.statoDesc ? (
+      <PossibleVoucherStateOption
+        text={listItem.item.statoDesc}
+        value={listItem.item.idStato}
+        checked={listItem.item.idStato === voucherStatus}
+        onPress={onVoucherStatusSelect}
+      />
+    ) : null;
+
+  const onRemoveButton = () => {
+    setSearchValue(undefined);
+    setVoucherStatus(undefined);
+    setDepartureDate(undefined);
+    setReturnDate(undefined);
+    props.resetFilter();
+    props.onClose();
+  };
+
+  const onConfirmButton = () => {
+    props.updateFilter({
+      codiceVoucher: searchValue,
+      idStato: voucherStatus,
+      dataDa: departureDate,
+      dataA: returnDate
+    });
+    props.onConfirm();
+  };
+
+  const selectedFilters =
+    (voucherStatus !== undefined ? 1 : 0) +
+    (departureDate !== undefined ? 1 : 0) +
+    (returnDate !== undefined ? 1 : 0) +
+    (searchValue && searchValue.length > 0 ? 1 : 0);
+
+  return (
+    <Container>
+      <AppHeader>
+        <Left />
+        <Body style={{ alignItems: "center" }}>
+          <H5 weight={"SemiBold"} color={"bluegrey"}>
+            {I18n.t("bonus.sv.voucherList.filter.title")}
+          </H5>
+        </Body>
+        <Right>
+          <ButtonDefaultOpacity onPress={props.onClose} transparent={true}>
+            <IconFont name="io-close" />
+          </ButtonDefaultOpacity>
+        </Right>
+      </AppHeader>
+      <SafeAreaView style={IOStyles.flex}>
+        <ScrollView
+          style={[IOStyles.flex]}
+          bounces={false}
+          keyboardShouldPersistTaps={"handled"}
+        >
+          <View style={IOStyles.horizontalContentPadding}>
+            <LabelledItem
+              iconPosition={"right"}
+              inputProps={{
+                value: searchValue,
+                onChangeText: setSearchValue,
+                placeholder: I18n.t(
+                  "bonus.sv.voucherList.filter.searchValuePlaceholder"
+                )
+              }}
+              icon="io-search"
+            />
```

Comment:
Atm this textnput doesn't have any size limit, perhaps it might be better to insert a maximum limit with the prop `maxLength`

Reply:
Added in [286793a](https://github.com/pagopa/io-app/commit/286793a84d672f021238af4540266a6f39f432b6)